shop/views.py

```diff
@@ -110,3 +111,35 @@ def get_context_data(self, **kwargs):
             id =  url[0]
             context['images'].append({'url': 'img/'+ value.url, 'id_img': id})
         return context
+
+
+def product_search(request):
+    
+    if request.POST:
+        query = request.POST['search_product']
+        request.session['search_product'] = query
+    elif request.session['search_product']:
+        query = request.session['search_product']
+    else:
+        query = ""
+    results = Product.objects.filter(name__icontains=query) |\
```

Comment:
xử lý validate `query` trước khi thực hiện truy vấn, vd: nhập toàn khoảng trắng, nhập các ký tự đặc biệt: %, #, ...

Reply:
chị ơi , phần này e xử lý được cái xóa hết khoảng trắng thôi ạ
ký tự đặc biệt em định dùng '[a-zA-Z0-9]*', nhưng mà tìm kiếm tiếng việt chữ đ,â ... nó cũng tính là ký tự đặc biệt nó xóa hết ạ

Reply:
- tìm kiếm nên cứ cho user nhập thoải mái em ạ, ko cần check regex chỗ này đâu, nhưng xử lý để kết quả trả về đúng. VD: khi nhập ký tự % thì tìm kiếm theo đúng ký tự %
- line 125, 126:
```
from django.db.models import Q
...
results = Product.objects.filter(Q(name__icontains=query) | Q(description__icontains=query))
```